### DISCUSSION TITLE
Qual linguagem, CMS ou Plataforma utilizar pra criar um site de compra e venda de moedas?

### DISCUSSION THREAD

**Discussion by ailtonrkr** [frontendbr/forum#1258] [Discussão]
Galera, alguém tem alguma idéia de como se construir um sistema de compra e venda de moedas online tipo a [confidenceCambio](https://www.confidencecambio.com.br/ecommerce/#/home?_k=i88g60)? 

Eu desenvolvo muito usando Wordpress, mas pelas minhas pesquisas não achei nada que de pra construir desse tipo usando o WP.

1 - Seria um sistema a ser desenvolvido em PHP? 
2 - Tentar algum SPA tipo Angular, React?
3 - Caso fosse possível construir usando WP, vocês acham que uma adaptada em plugins de e-commerce tipo o woocommerce atenderia minha necessidade?

Enfim, esse é o projeto mais casca grossa que peguei até hoje e meu foco é mais a construção de layouts mesmo, então tô meio numa sinuca de bico pra resolver isso ai, se alguém tiver experiência com algo do tipo e puder me dar uma luz vai ser massa.

**Comment by waldemirflj** [comment]
@ailtonrkr tempos atrás (2010) participei do desenvolvimento de uma plataforma de leilão de um centavo. Ná época não foi tão tranquilo desenvolver isso, ainda mais do zero. A maior barreira foi escrever o algoritmo de conversão, existiam muitas regras e funcionalidades customizadas. 

Recomendo você começar pelo backend da plataforma, comece desenhando os cenários, fluxos e funcionalidades. O frontend é importante, mas vejo isso como último passo.

Sobre a linguagem, isso é bem pessoal de cada um (que não seja xiita). Se você sabe PHP, faça com PHP. Se amanhã a tecnologia não ter atender mais, você muda, sempre que eu começo um projeto tento visualizar o MVP da coisa, com o tempo vou atualizando.

**Comment by luizbills** [comment]
> Enfim, esse é o projeto mais casca grossa que peguei até hoje e meu foco é mais a construção de layouts mesmo, então tô meio numa sinuca de bico pra resolver isso ai, se alguém tiver experiência com algo do tipo e puder me dar uma luz vai ser massa.

Minha primeira recomendação é: evita aceitar trabalhos que você não dá conta sozinho. Ou se mesmo assim quiser pegar e arriscar, tercerize para um colega desenvolvedor(a) de sua confiança.

Sobre a tecnologia do projeto, dá para fazer com WordPress sim. Você pode aproveitar o woocommerce para não ter que desenvolver a integração com os gateways de pagamento. Porém você vai ter que dar uma customizada para tirar algumas funcionalidades de loja virtual.

**Comment by kavalcante** [comment]
Honestamente, isso não é um projeto para ser feito em cima de uma plataforma comum (a.k.a Wordpress). A não ser que isso seja para fins de MVP e pouca grana investida. Esse cenário mostrado, é de uma plataforma complexa, que possui regras de negócios específicas e pode variar muito dependendo do cenário.

Sobre linguagem, não acredito que isso seja um ponto extremamente importante nesse primeiro momento. Eu faria na stack que estou mais acostumado, e que me dê uma possibilidade de moldar as regras de negócio como eu quiser.

Aqui usamos Node.js com Express. Mas pode ser em PHP com Laravel, Ruby com Rails, Python com Django ou qualquer outra coisa... Agora eu não iria de CMS, como Wordpress. Pq acho que não é pra isso que ele foi feito. Tu precisa construir visualizações internas e telas que são específicas desse negócio...

Outro ponto importante, você precisa de ajuda nesse projeto. Tanto de conhecimento do negócio (o contratante deve te ajudar), além de possível ajuda técnica mesmo. Com outras pessoas envolvidas.
